Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift

```diff
@@ -215,12 +220,14 @@ final class ApproveEngine {
 private extension ApproveEngine {
 
     func setupRequestSubscriptions() {
-        pairingRegisterer.register(method: SessionProposeProtocolMethod())
+        pairingRegisterer.register(method: SessionProposeProtocolMethod.responseAutoReject())
             .sink { [unowned self] (payload: RequestSubscriptionPayload<SessionType.ProposeParams>) in
                 guard let pairing = pairingStore.getPairing(forTopic: payload.topic) else { return }
+                let responseApproveMethod = SessionAuthenticatedProtocolMethod.responseApprove().method
```

Comment:
also feels strange that u need to init with responseApprove() to just get a method string

Reply:
☝️ https://github.com/WalletConnect/WalletConnectSwiftV2/pull/1364#discussion_r1619024300